src/native/corehost/fxr/fx_muxer.cpp

```diff
@@ -1018,21 +1019,68 @@ int fx_muxer_t::handle_exec_host_command(
         required_buffer_size);
 }
 
+namespace
+{
+    pal::architecture get_requested_architecture(int argc, const pal::char_t* argv[])
+    {
+        // Default to current architecture if architecture is not specified
+        if (argc < 3 || pal::strcasecmp(_X("--arch"), argv[2]) != 0)
```

Comment:
I would change this to assume the inputs are right where they should be. See my other comment at the callsite. The goal here would be to verify `argc` is `2` - the amount needed by the function, and `argv[0] == "--arch"` and `argv[1] == "<arch>"`. That would make this function very clear about expectations and avoid some of these unnatural indices.